eventlog/service.go

```diff
@@ -81,9 +86,43 @@ func (s *Service) Log(req *LogRequest) (*LogResponse, error) {
 	return &LogResponse{}, nil
 }
 
+// GetEvent asks omniledger for a stored event.
+func (s *Service) GetEvent(req *GetEventRequest) (*GetEventResponse, error) {
+	req2 := omniledger.GetProof{
+		Version: omniledger.CurrentVersion,
+		Key:     req.Key,
+		ID:      req.SkipchainID,
+	}
+	reply, err := s.omni.GetProof(&req2)
+	if err != nil {
+		return nil, err
+	}
+	if !reply.Proof.InclusionProof.Match() {
+		return nil, errors.New("not an inclusion proof")
+	}
+	k, vs, err := reply.Proof.KeyValue()
+	if err != nil {
+		return nil, err
+	}
+	if !bytes.Equal(k, req2.Key) {
+		return nil, errors.New("wrong key")
+	}
+	if len(vs) < 2 {
+		return nil, errors.New("not enough values")
+	}
+	e := Event{}
+	err = protobuf.Decode(vs[0], &e)
+	if err != nil {
+		return nil, err
+	}
+	return &GetEventResponse{
+		Event: e,
+	}, nil
+}
+
 const contractName = "eventlog"
 
-func (s *Service) decodeAndCheckEvent(eventBuf []byte) (*Event, error) {
+func (s *Service) decodeAndCheckEvent(coll collection.Collection, eventBuf []byte) (*Event, error) {
 	// Check the timestamp of the event: it should never be in the future,
```

Comment:
The timestamp _might_ be in the future, because the node's clock is running too slow! So I think we really need to accept a time-window here that is independent of the `blockinterval`. This should probably be a configuration parameter of the event-logger. Something like 10 minutes or so.

The alternative is if the client-instruction creates the timestamp itself, then we could be a bit more rigid to what timestamps we accept.

Reply:
Your suggestion sounds reasonable, I'll note it down and make an issue on cothority when we merge.